Formula/t/technitium-dns.rb

```diff
@@ -0,0 +1,68 @@
+class TechnitiumDns < Formula
+  desc "Self host a DNS server for privacy & security"
+  homepage "https://technitium.com/dns/"
+  url "https://github.com/TechnitiumSoftware/DnsServer/archive/refs/tags/v13.5.0.tar.gz"
+  sha256 "7a5f47af4c938c5983e415547b777f853ede5ada8837012c8b5e8e36df1a380e"
+  license "GPL-3.0-or-later"
+
+  depends_on "dotnet@8"
+  depends_on "libmsquic"
+
+  on_linux do
+    depends_on "bind" => :test # for `dig`
+  end
+
+  resource "TechnitiumLibrary" do
+    url "https://github.com/TechnitiumSoftware/TechnitiumLibrary/archive/refs/tags/dns-server-v13.5.0.tar.gz"
+    sha256 "8e0f3200ad6ee6a1ab346a7224a1bc538f6b3dc1b660cd8bb725d0dc09fa2bf9"
+  end
+
+  def install
+    # See https://github.com/TechnitiumSoftware/DnsServer/blob/master/build.md#build-instructions
+    dotnet = Formula["dotnet@8"]
+    mkdir_p "DnsServer"
+    mv Dir.glob("*").reject { |f| f == "DnsServer" }, "DnsServer"
+    resource("TechnitiumLibrary").stage do
+      system dotnet.opt_bin/"dotnet", "build",
+             "TechnitiumLibrary.ByteTree/TechnitiumLibrary.ByteTree.csproj",
+             "-c", "Release"
+      system dotnet.opt_bin/"dotnet", "build",
+             "TechnitiumLibrary.Net/TechnitiumLibrary.Net.csproj",
+             "-c", "Release"
+      cp_r ".", buildpath/"TechnitiumLibrary"
+    end
+    system dotnet.opt_bin/"dotnet", "publish", "DnsServer/DnsServerApp/DnsServerApp.csproj", "-c", "Release"
```

Comment:
The declared dependency should already take care of the correct dotnet. Also adjusted the cp to be easier to read.
```suggestion
      system "dotnet", "build",
             "TechnitiumLibrary.ByteTree/TechnitiumLibrary.ByteTree.csproj",
             "-c", "Release"
      system "dotnet", "build",
             "TechnitiumLibrary.Net/TechnitiumLibrary.Net.csproj",
             "-c", "Release"
      (buildpath/"TechnitiumLibrary").install Dir.glob("*")
    end
    system "dotnet", "publish", "DnsServer/DnsServerApp/DnsServerApp.csproj", "-c", "Release"
```